OpenSim/Actuators/Thelen2003Muscle.cpp

```diff
@@ -762,110 +759,161 @@ Thelen2003Muscle::initMuscleState(const SimTK::State& s,
+    partialsFunc();
 
-                if(lce < getPennationModel().getMinimumFiberLength())
-                {
-                    minFiberLengthCtr++;
-                    lce = getPennationModel().getMinimumFiberLength();
-                }
+    // Update the velocity (and multiplier) estimate from partials
+    velocityFunc();
 
-                
-                //Update position level quantities, only if they won't go 
-                //singular
-
-                phi = getPennationModel().calcPennationAngle(lce);
-                cosphi = cos(phi);
-                tl  = ml - lce*cosphi;
-                lceN = lce/ofl;
-                tlN  = tl/tsl;
-
-                //Update velocity level quantities: share the muscle velocity 
-                //between the tendon and the fiber according to their relative 
-                //stiffness:
-                //
-                //Fm = Ft at equilibrium
-                //Fm = Km*lceAT
-                //Ft = Kt*xt
-                //dFm_d_xm = Km*dlceAT + dKm_d_t*lceAT (assume dKm_d_t = 0)
-                //dFt_d_xt = Kt*dtl + dKt_d_t*dtl (assume dKt_d_t = 0)
-                //
-                //This is a heuristic. The above assumptions are necessary as 
-                //computing the partial derivatives of Km or Kt w.r.t. requires 
-                //acceleration level knowledge, which is not available in 
-                //general.
-
-                //Stiffness of the muscle is the stiffness of the tendon and the 
-                //fiber (along the tendon) in series
-
-                //the if statement here is to handle the special case when the
-                //negative stiffness of the fiber (which could happen in this
-                // model) is equal to the positive stiffness of the tendon.
-                if( abs(dFmAT_dlceAT + dFt_d_tl) > SimTK::SignificantReal
-                    && tl > getTendonSlackLength()){
-                    Ke      = (dFmAT_dlceAT*dFt_d_tl)/(dFmAT_dlceAT + dFt_d_tl); 
-                    dtl     = (1/dFt_d_tl)*Ke*dml;
-                }else{
-                    dtl     = dml;
-                }
+    // Compute the equilibrium error with velocity estimate
+    ferrFunc();
+
+    // Update the partial derivatives of the force error w.r.t. lce with 
+    // newly estimated fv
+    partialsFunc();
 
-                dlce = getPennationModel().calcFiberVelocity(cosphi,dml,dtl);
-                dlceN    = dlce/(vmax*ofl);
-                dphi = getPennationModel().calcPennationAngularVelocity(
-                                           tan(phi),lce,dlce);
-                /*dlceAT = */getPennationModel().calcFiberVelocityAlongTendon(
-                                             lce,dlce,sinphi,cosphi,dphi);
+    double ferrPrev = ferr;
+    double lcePrev = lce;
+
+    double h = 1.0;
+    while( (abs(ferr) > aSolTolerance)  && (iter < aMaxIterations)) {
+        dferr_d_lce = dFmAT_dlce - dFt_d_lce;
```

Comment:
Added the comment and update the remainder of comments so they correspond identically (or so I hope).